docs/migratation-guide-v1.0.rst

```diff
@@ -0,0 +1,311 @@
+to work but it’s still highly recommended to test your application
+first.
+
+.. WARNING::
+   **BREAKING CHANGES**
+
+BREAKING CHANGES
+================
+
+``iConn.run()`` no longer supports sync mode
+--------------------------------------------
+
+Sync mode did not work properly to begin with see
+(`#32 <https://github.com/IBM/nodejs-itoolkit/issues/32>`__).
+
+``iConn.setTimeout()`` is removed
+----------------------------------
+
+This function was used to set timeout for ``iConn.run`` sync mode.
+
+``iSql.connect()`` and ``iSql.setOptions()`` are removed
+--------------------------------------------------------
+
+These functions were used in conjunction for XMLSERVICE user
+authentication. The transports already handle user authentication.
+
+New Features
+============
+
+SSH and ODBC transports
+-----------------------
+
+Users can now use ``ssh`` and ``odbc`` transports. This will allow users
+more ways to use itoolkit on their local machine.
+
+Support error first callbacks
+-----------------------------
+
+-  ``iConn.run()`` did not return errors to the run callback.
+   ``Connection.run()`` follows Node.js convention of `error first
+   callbacks <https://nodejs.org/api/errors.html#errors_error_first_callbacks>`__.
+   ``Connection`` still has a compatability option ``returnError`` to
+   behave like ``iConn.run()`` and return the xml output as the first
+   parameter of the run callback.
+
+
+Support DS types within addReturn
+---------------------------------
+
+-  ``iPgm.addReturn`` did not support the DS data type,
+   ``ProgramCall.addReturn`` added support.
+
+Deprecated Classes and Functions
+================================
+
+iConn
+-----
+
+The ``iConn`` class was replaced with the ``Connection`` class and it
+will be removed in ``v2.x``.
+
+Differences
+^^^^^^^^^^^
+
+-  ``Connection`` constructor accepts single object parameter.
+-  ``Connection.run()`` follows Node.js convention of error first `error
+   first
+   callbacks <https://nodejs.org/api/errors.html#errors_error_first_callbacks>`__.
+
+Migrating from ``iConn`` to ``Connection`` Constructor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: js
+
+   // using idb transport
+
+   // const conn = new iConn("*LOCAL", "myuser", "mypassword");
+
+   const conn = new Connection({
+     transport: 'idb',
+     transportOptions: { database: '*LOCAL', username: 'myuser', password: 'mypassword' }
+   });
+
+   // using rest transport
+
+   // const restConfig = { host: 'myhost', port: 80, path: '/' }
+   // const conn = new iConn('*LOCAL', 'myuser', 'mypassword', restConfig);
+
+   const conn = new Connection({
+     transport: 'rest',
+     transportOptions: {
+       database: '*LOCAL',
+       username: 'myuser',
+       password: 'mypassword',
+       url: 'http://myhost:80/',
+     }
+   });
+
+Migrating from ``iConn.run()`` to ``Connection.run()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This section is way more important to highlight than the prior one. You should be able to figure out how to connect using Connection from the docs, however you may not realize that iConn -> Connection implies a change in behavior of the callbacks. I think we definitely need some WARNING messages above to point the user at this section and mention `returnError`.


Something like:

"Beware that the Connection and iConn classes differ in how they call the callbacks passed to their run methods. You cannot simply replace iConn with Connection without adjusting your callbacks. See here for more info on how to do that.